From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: TominoLP <erfinder.tom@gmail.com>
Date: Mon, 13 Mar 2023 19:43:46 +0100
Subject: [PATCH] rebrand


diff --git a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
index 95c4bc1a58aa7e447f62d5553111d9967cbd3cba..39e31ab4a4b4f52c508081601c987f7ce1636902 100644
--- a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
+++ b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
@@ -33,8 +33,8 @@ public class PaperVersionFetcher implements VersionFetcher {
     @Nonnull
     @Override
     public Component getVersionMessage(@Nonnull String serverVersion) {
-        String[] parts = serverVersion.substring("git-Plazma-".length()).split("[-\\s]"); // Purpur // Plazma
-        final Component updateMessage = getUpdateStatusMessage("PlazmaMC/Plazma", "ver/" + getMinecraftVersion(), parts[0]); // Purpur // Plazma
+        String[] parts = serverVersion.substring("git-Wave-".length()).split("[-\\s]"); // Purpur // Plazma
+        final Component updateMessage = getUpdateStatusMessage("TominoLP/Wave", "ver/" + getMinecraftVersion(), parts[0]); // Purpur // Plazma
         final Component history = getHistory();
 
         return history != null ? Component.join(net.kyori.adventure.text.JoinConfiguration.separator(Component.newline()), history, updateMessage) : updateMessage; // Purpur
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 97af3b01e1c30f7a286122794d9ecbd26988f471..214ff252c2b220051083b10b838f55256bd898af 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -261,7 +261,7 @@ import javax.annotation.Nullable; // Paper
 import javax.annotation.Nonnull; // Paper
 
 public final class CraftServer implements Server {
-    private final String serverName = "Plazma"; // Paper // Purpur // Plazma
+    private final String serverName = "Wave"; // Paper // Purpur // Plazma // Wave
     private final String serverVersion;
     private final String bukkitVersion = Versioning.getBukkitVersion();
     private final Logger logger = Logger.getLogger("Minecraft");
diff --git a/src/main/java/org/purpurmc/purpur/PurpurConfig.java b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
index d254467cd806ed2a650ff1e66eb4c914f545adf7..12488af544dd587a2f43caa116e5603be9fddbd1 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
@@ -227,7 +227,7 @@ public class PurpurConfig {
         deathMessageOnlyBroadcastToAffectedPlayer = getBoolean("settings.broadcasts.death.only-broadcast-to-affected-player", deathMessageOnlyBroadcastToAffectedPlayer);
     }
 
-    public static String serverModName = "Plazma"; // Plazma
+    public static String serverModName = "Wave"; // Plazma // Wave
     private static void serverModName() {
         serverModName = getString("settings.server-mod-name", serverModName);
     }
